resources-manager/include/AdaptivePruningManager.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "IRequestHandler.hpp"
+#include "IResourceManager.hpp"
+#include "InternalBFTClient.hpp"
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+enum PruningMode { LEGACY, ADAPTIVE };
```

Comment:
What about MANUAL mode, which I remember from the design document? Was that dropped?

Reply:
It is very similiar to the legacy mode, so maybe it will not be needed at all.

Reply:
Besides enabling the thread, do those flags have any other meaning?